src/modules/common/Config.ts

```diff
@@ -136,9 +137,13 @@ export class ServerConfig implements IServerConfig
      */
     constructor (address?: string, port?: number, agora_endpoint?: string)
     {
-        let conf = extend(true, {}, ServerConfig.defaultValue());
+        let default_config = ServerConfig.defaultValue();
+        let conf = extend(true, {}, default_config);
         extend(true, conf, {address: address, port: port, agora_endpoint: agora_endpoint});
-        this.address = conf.address;
+        if (ip.isV4Format(conf.address) || ip.isV6Format(conf.address))
+            this.address = conf.address;
+        else
+            this.address = default_config.address;
```

Comment:
No! If the user enters wrong data, we *need* to inform them, no silently default to something else.
Having sensible defaults is important, but we should, *never*, *ever*, silently ignore wrong input from the user, it just creates an absolutely horrible user experience.

Reply:
Yeah, that's right. That's a good idea.
I think I want automatic too much.

Reply:
When the wrong IP address was entered, I changed the process to output the log and exit.